.gitlab/shell.nix

```diff
@@ -0,0 +1,90 @@
+{ system ? "aarch64-darwin"
```

Comment:
I note that this nix script is only used for the aarch64-darwin config, therefore I would change the file name to `shell-aarch64-darwin.nix` or similar